luarules/gadgets/unit_gunship_strafe_fix.lua

```diff
@@ -0,0 +1,100 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+	return {
+		name = "Gunship Strafe Fix",
+		desc = "When gunships stop attacking their target unit with no followup command, forces them to stop strafing and enter idle behavior",
+		author = "JRTaylord: https://github.com/JRTaylord",
+		date = "May 7, 2025",
+		license = "GNU GPL, v2 or later",
+		layer = 0, -- Todo: ask which layer should this be on
+		enabled = true,
+	}
+end
+
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+local spGetUnitWeaponTarget = Spring.GetUnitWeaponTarget
+local spGetUnitCommands = Spring.GetUnitCommands
+local spGetAllUnits = Spring.GetAllUnits
+local spGetUnitDefID = Spring.GetUnitDefID
+local spEcho = Spring.Echo
+
+-- Constants
+-- Todo figure out options to make the command silent
+local CMD_OPTIONS = CMD.OPT_INTERNAL
+
+-- Gadget variables
+local gunshipDefs = {}
+local gunshipsToTrack = {}
+
+-- Only run in synced code
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+-- Helper functions
+local function isTargettingUnit(unitID, unitDefID)
+	local weaponCount = gunshipDefs[unitDefID].weaponCount
+	for weaponNum = 1, weaponCount do
+		local targetType, _, targetID = spGetUnitWeaponTarget(unitID, weaponNum)
+		if targetType == 1 and targetID then
+			return true
+		end
+	end
+	return false
+end
+
+local function stopGunshipIfNoCmdAndTarget(unitId, unitDefID)
+	local numCommands = #spGetUnitCommands(unitID, 1)
+	local hasTarget = isTargettingUnit(unitID, unitDefID)
+	if numCommands == 0 and not hasTarget then
+		spGiveOrderToUnit(unitID, CMD.STOP, {}, CMD_OPTIONS)
+	end
+end
+
+function gadget:Initialize()
+	-- Find all gunship units
+	for unitDefID, unitDef in pairs(UnitDefs) do
+		if
+			(unitDef.canFly or unitDef.isAirUnit)
+			and (unitDef.gunship or unitDef.isHoveringAirUnit or unitDef.hoverattack)
+			and #unitDef.weapons > 0
+		then
+			gunshipDefs[unitDefID] = {
+				name = unitDef.name,
```

Comment:
Doesn't look used, perhaps just keep the weapon count instead of a table?

Reply:
Yeah, this was for logging and I'll be cutting it from the table